lib/galaxy/tool_util/linters/inputs.py

```diff
@@ -1,6 +1,7 @@
 """This module contains a linting functions for tool inputs."""
+from galaxy.tools.parameters.basic import ToolParameter
```

Comment:
Importing from `galaxy.tools` inside `galaxy.tool_util` would break the splitting in packages. I'd move the `parse_name()` method from `galaxy.tools.parameters.basic.ToolParameter` to `galaxy.tool_util.parser.xml.XmlInputSource` . Ping @jmchilton for double checking the idea.